src/cli/simple/command/mod.rs

```diff
@@ -821,9 +846,17 @@ impl ContractCall<'_> {
             FinalExecutionStatus::Failure(e) => {
                 anyhow::bail!("{}: {e}", self.error_message)
             }
-            FinalExecutionStatus::SuccessValue(_) => {
-                println!("{}", self.success_message);
-            }
+            FinalExecutionStatus::SuccessValue(_) => match context.output_format {
+                OutputFormat::Plain => println!("{}", self.success_message),
+                OutputFormat::Json => {
+                    let formatted = to_string_pretty(&result.transaction_outcome)?;
+                    println!("{formatted}");
+                }
+                OutputFormat::Toml => {
+                    let formatted = toml::to_string_pretty(&result.transaction_outcome)?;
+                    println!("{formatted}");
+                }
+            },
```

Comment:
I would recommend moving the logic of the creation message into a separate function. Also, I thought about the creation of an output with some specific fields rather than serialising the outcome, but it could be done in another PR.

Reply:
@aleksuss  To my taste, there is not too much code in a single function now, so I prefer to leave it the way it is now, to avoid introducing a new indirection layer (aka some other function to format messages). With more requirements for a formatter we can introduce a separate module `formatter` to hold transformation and display logic and put everything there. 

For this PR I'd leave it the way it is now, but I'm open to changing it if you think it is better!